### PR TITLE
Add blocking ability

### PR DESCRIPTION
**Pull request by jrosen081**
image

### CODE REVIEW

File: frontend/src/components/Blocker.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    children: ReactElement,
+    afterSeconds: number,
+    message: string
+}
+
+const Blocker: FunctionComponent<BlockerProps> = ({ afterSeconds, children, message }) : ReactElement => {
+    const [lastTap] = useState(new Date());
+    const [currentTime, setCurrentTime] = useState(new Date());
+    const [hasShown, setHasShown] = useState(false);
+    let onClick = children.props.onClick;
```
Comment (cindy1u0) [inline frontend/src/components/Blocker.tsx:15]: I assume this onClick is for the next button

File: frontend/src/components/Blocker.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    }
+    useEffect(() => {
+        let interval = window.setInterval(() => {
+            setCurrentTime(new Date())
+        }, 1000)
+        return () => clearInterval(interval);
+    }, [])
+    const timeSince = currentTime.getTime() - lastTap.getTime();
+    let shouldShowBlocker = timeSince >= afterSeconds * 1000;
+
```
Comment (cindy1u0) [inline frontend/src/components/Blocker.tsx:34]: Is it possible to make it repeatable? So basically like onOk, if the nextButton is still not clicked, and after the same amount of time, show it again

**Review by cindy1u0 — COMMENTED** [review]
Looks good otherwise :)

File: frontend/src/App.tsx [hunk 3]
```diff
@@ -17,7 +18,8 @@ const App = () : ReactElement => {
         <Route exact path="/login" component={Login} />
         <Route path="/sign_up" component={CreateUser} />
         <AuthenticatedRoute path="/dashboard" component={Dashboard} />
-        <AuthenticatedRoute path="/interventions" component={Interventions} />
+        <AuthenticatedRoute exact path="/interventions" component={Interventions} />
+        <AuthenticatedRoute path="/interventions/1" component={FirstActivity} />
```
Comment (cindy1u0) [inline frontend/src/App.tsx]: Prob get rid of this later

File: frontend/src/components/Blocker.tsx (hunk 1 above)
Comment (jrosen081) [reply frontend/src/components/Blocker.tsx:15]: Yep. This is so the click events go through

File: frontend/src/components/Blocker.tsx (hunk 2 above)
Comment (jrosen081) [reply frontend/src/components/Blocker.tsx:34]: I will work on this, but it _should_ be possible

File: frontend/src/App.tsx (hunk 3 above)
Comment (jrosen081) [reply frontend/src/App.tsx]: Oh true lol. Will do

**Pull request merged**